PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -112,13 +150,24 @@ namespace // anonymous
 
 	generic_string getDialogFileName(IFileDialog* dialog)
 	{
+		generic_string fileName;
 		PWSTR pszFilePath = nullptr;
-		dialog->GetFileName(&pszFilePath);
-		generic_string fileName = pszFilePath;
+		HRESULT hr = dialog->GetFileName(&pszFilePath);
+		if (SUCCEEDED(hr) && pszFilePath)
+			fileName = pszFilePath;
 		CoTaskMemFree(pszFilePath);
```

Comment:
Should `CoTaskMemFree(pszFilePath);` be in the scope of `if (SUCCEEDED(hr) && pszFilePath)` ?

Reply:
To me, it seems safer this way. Besides, the function works with null just fine. [MSDN](https://docs.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-cotaskmemfree):
> A pointer to the memory block to be freed. If this parameter is NULL, the function has no effect.

Reply:
> To me, it seems safer this way. 

My point is, in the old implementation, if `! SUCCEEDED(hr)`, `pszFilePath` won't be allocated, but you still free it.

> Besides, the function works with null just fine.

OK - good to know.